File: data/readable_format/fortran/file2000.f90
! Main program and subroutine in the same file
PROGRAM TestCHKRNG
  IMPLICIT NONE
  INTEGER, PARAMETER :: N = 5
  INTEGER :: IARRAY(N) = (/ 0, 2, 5, 10, 15 /)
  INTEGER :: NITEMS, MAXVAL, NZERO, NERR

  ! Initialize variables
  NITEMS = N
  MAXVAL = 10
  ! Call the subroutine
  CALL CHKRNG(IARRAY, NITEMS, MAXVAL, NZERO, NERR)

  ! Output results
  PRINT *, 'NZERO = ', NZERO
  PRINT *, 'NERR = ', NERR
END PROGRAM TestCHKRNG

! Subroutine definition
SUBROUTINE CHKRNG (IARRAY, NITEMS, MAXVAL, NZERO, NERR)
  INTEGER :: IARRAY(*), NITEMS, MAXVAL, NZERO, NERR, I, N

  NERR  = 0
  NZERO = 0
  DO I = 1, NITEMS
     N = IARRAY(I)
     IF (N .LT. 1) THEN
        NZERO = NZERO + 1
     END IF
     IF (N .GT. MAXVAL) THEN
        NERR  = NERR + 1
     END IF
  END DO
END SUBROUTINE CHKRNG

File: data/readable_format/cpp/file2000.cpp
#include <vector>

void checkRange(const std::vector<int>& iArray, int nItems, int maxVal, int& nZero, int& nErr) {
    // Initialize counters
    nZero = 0;
    nErr = 0;

    // Loop through the array elements
    for (int i = 0; i < nItems; i++) {
        int n = iArray[i];
        if (n < 1) {
            nZero++;
        }
        if (n > maxVal) {
            nErr++;
        }
    }
}